src/main/java/io/jenkins/plugins/gitlabbranchsource/WebhookRegistrationTrait.java

```diff
@@ -58,7 +72,9 @@ public final GitLabWebhookRegistration getMode() {
      */
     @Override
     protected void decorateContext(SCMSourceContext<?, ?> context) {
-        ((GitLabSCMSourceContext) context).webhookRegistration(getMode());
+        GitLabSCMSourceContext ctx = (GitLabSCMSourceContext) context;
+        ctx.webhookRegistration(getMode());
+        ctx.withSystemHooksDisabled(isAdmin());
```

Comment:
Hmmm seems like we have logic bug :laughing: 

when isAdmin is true it turns systemHooksDisabled to true which goes ahead creating the system hook even though systemHooksDisabled was set to true.

Perhaps best to rename it `isAdmin` and `systemHooksDisabled` to `systemHooksEnabled` to avoid confusion.

Reply:
Yeah the literals used here are confusing. I need to fix this.